From cb761412e299ef907f22cd7c4146d50c8a792003 Mon Sep 17 00:00:00 2001
From: Guiyao <guiyao@huawei.com>
Date: Mar 26 2020 17:54:17 +0000
Subject: Unlock cond_mutex before pthread exit in gp_worker_main()


Signed-off-by: GuiYao <guiyao@huawei.com>
[rharwood@redhat.com: whitespace, tweak commit message]
Reviewed-by: Robbie Harwood <rharwood@redhat.com>

---

diff --git a/src/gp_workers.c b/src/gp_workers.c
index 18f38f6..ae42cef 100644
--- a/src/gp_workers.c
+++ b/src/gp_workers.c
@@ -369,6 +369,7 @@ static void *gp_worker_main(void *pvt)
             /* wait for next query */
             pthread_cond_wait(&t->cond_wakeup, &t->cond_mutex);
             if (t->pool->shutdown) {
+                pthread_mutex_unlock(&t->cond_mutex);
                 pthread_exit(NULL);
             }
         }
